core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java

```diff
@@ -118,31 +128,33 @@ public GeyserServer(GeyserImpl geyser, int threadCount) {
     }
 
     public CompletableFuture<Void> bind(InetSocketAddress address) {
-        CompletableFuture<Void> future = new CompletableFuture<>();
-        this.bootstrapFuture = this.bootstrap.bind(address).addListener(bindResult -> {
-            if (bindResult.cause() != null) {
-                future.completeExceptionally(bindResult.cause());
-                return;
-            }
-            future.complete(null);
-        });
+        bootstrapFutures = new ChannelFuture[listenCount];
+        for (int i = 0; i < listenCount; i++) {
+            bootstrapFutures[i] = Optional.of(bootstrap)
+                    .map(f -> f.bind(address))
+                    .map(f -> addHandlers(f, this))
+                    .orElseThrow();
+        }
 
-        Channel channel = this.bootstrapFuture.channel();
+        return Bootstraps.allOf(bootstrapFutures);
+    }
 
+    private static ChannelFuture addHandlers(ChannelFuture f, GeyserServer self) {
```

Comment:
`future` for the 1st parameter. Can this be an instance method? Otherwise prefer `server` for the 2nd